src/WebJobs.Script/Workers/Rpc/WebHostRpcWorkerChannelManager.cs

```diff
@@ -206,15 +206,22 @@ private bool UsePlaceholderChannel(IRpcWorkerChannel channel)
                 return versionMatches;
             }
 
-            // Special case: node and PowerShell apps must be read-only to use the placeholder mode channel
+            // Special case: node, python and PowerShell apps must be read-only to use the placeholder mode channel
             // Also cannot use placeholder worker that is targeting ~3 but has backwards compatibility with V2 enabled
             // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
             if (string.Equals(workerRuntime, RpcWorkerConstants.NodeLanguageWorkerName, StringComparison.OrdinalIgnoreCase)
                 || string.Equals(workerRuntime, RpcWorkerConstants.PowerShellLanguageWorkerName, StringComparison.OrdinalIgnoreCase)
                 || string.Equals(workerRuntime, RpcWorkerConstants.PythonLanguageWorkerName, StringComparison.OrdinalIgnoreCase))
             {
                 // Use if readonly and not v2 compatible on ~3 extension
-                return _applicationHostOptions.CurrentValue.IsFileSystemReadOnly && !_environment.IsV2CompatibleOnV3Extension();
+                var isReadOnlyAndNoCompat = _applicationHostOptions.CurrentValue.IsFileSystemReadOnly && !_environment.IsV2CompatibleOnV3Extension();
+
+                if (!isReadOnlyAndNoCompat)
+                {
+                    _logger.LogInformation("App will not use placeholder channel - ReadOnly: {isReadOnly}. NoCompat: {noCompat}.", _applicationHostOptions.CurrentValue.IsFileSystemReadOnly, !_environment.IsV2CompatibleOnV3Extension());
```

Comment:
nit: change to LogDebug

Reply:
Any reason why? Given that this is impacting behavior (not using channel), I think information is correct.

Reply:
For Kusto debug is default level; LogInformation, is default level for app insights is information; My recommendation anything "System Logs" that does not have useful information for Cx, should be debug

Reply:
Done.